tests/robustness/validate/validate.go

```diff
@@ -71,7 +71,7 @@ func mergeWatchEventHistory(reports []report.ClientReport) ([]model.WatchEvent,
 						if prev, found := revisionToEvents[lastRevision]; found {
 							// This assumes that there are txn that would be observed differently by two watches.
 							// TODO: Implement merging events from multiple watches about single revision based on operations.
-							if diff := cmp.Diff(prev.events, events); diff != "" {
+							if diff := cmp.Diff(prev.events, events, cmp.Comparer(compareWatchEvents)); diff != "" {
```

Comment:
It would be be simpler and cleaner for this function to track and return `[]model.PersistedEvent` instead of rewriting comparer to skip everything apart of `PersistedEvent` field.

Reply:
You're right - it would be cleaner. The reason for me doing this is: https://github.com/etcd-io/etcd/pull/17352#discussion_r1486491484

Would you prefer patching it later before it is passed into `validatePrevKV`?